engine/src/main/java/org/hibernate/validator/internal/engine/ValidationContext.java

```diff
@@ -275,18 +274,17 @@ public void addConstraintFailures(Set<ConstraintViolation<T>> failingConstraintV
 				messageTemplate,
 				localContext.getCurrentValidatedValue(),
 				descriptor,
+				constraintViolationCreationContext.getMessageParameters(),
 				constraintViolationCreationContext.getExpressionVariables()
 		);
 		// at this point we make a copy of the path to avoid side effects
 		Path path = PathImpl.createCopy( constraintViolationCreationContext.getPath() );
-		//same for expression variables
-		Map<String, Object> expressionVariables =
-				Collections.unmodifiableMap( constraintViolationCreationContext.getExpressionVariables() );
```

Comment:
We don't need a copy here anymore because the maps in constraintViolationCreationContext are now immutable so we don't allow the side effects that might have existed before and have justified this copy.